server/src/main/java/org/elasticsearch/common/xcontent/ToXContent.java

```diff
@@ -132,4 +130,24 @@ public Boolean paramAsBoolean(String key, Boolean defaultValue) {
     default boolean isFragment() {
         return true;
     }
+
+    /**
+     * Parse {@code value} with values "true", "false", or null, returning the
+     * default value if null or the empty string is used. Any other input
+     * results in an {@link IllegalArgumentException} being thrown.
+     */
+    static boolean parseBoolean(String value, Boolean defaultValue) {
```

Comment:
Would you move this to a package-visible class named `Booleans`? There is no confusion with the other `Booleans` because of package visibility. It does not make sense on this class, it does not make sense on the other class, and the name will help for ease of finding in the future if needed.

Reply:
Certainly, I'll do that.